### ISSUE TITLE
Cannot build on GHC 8

### ISSUE THREAD

**Issue by ocharles** [paul-rouse/mysql#20]
```
Setup.lhs:39:41: error:
    • No instance for (ConstOrId
                         (IO (Maybe FilePath))
                         (ProgramSearchPath -> IO (Maybe (FilePath, [FilePath]))))
        arising from a use of ‘constOrId’
        (maybe you haven't applied a function to enough arguments?)
    • In the expression:
        constOrId
        $ liftM2
            mplus
            (findProgramLocation verbosity "mysql_config")
            (findProgramLocation verbosity "mysql_config5")
      In the ‘programFindLocation’ field of a record
      In the expression:
        (simpleProgram "mysql_config")
          {programFindLocation = \ verbosity
                                   -> constOrId
                                      $ liftM2
                                          mplus
                                          (findProgramLocation verbosity "mysql_config")
                                          (findProgramLocation verbosity "mysql_config5")}
```

**Comment by jwaldmann** [comment]
I have the same error with ghc-7.10.3. 
I think this is not about ghc version, but about Cabal, or cabal-install.

I built cabal-install like this: https://mail.haskell.org/pipermail/ghc-devs/2016-April/011751.html
and then I could build the mysql package.

**Comment by LukaHorvat** [comment]
So what's the deal with this? I still can't get it building. Same error.

**Comment by charlydagos** [comment]
Same issue here

**Comment by mchaver** [comment]
It seems to not compile because of Cabal 1.24.0. I ran into the same issue with ghc-7.10.3. https://github.com/bos/mysql/pull/21 did not seem to fix it for 7.10.3. I made another pull request that hopefully works for 7 and 8 https://github.com/bos/mysql/pull/22

**Comment by defanor** [comment]
With ghc-7.10.3, it didn't work for me with cabal 1.24, or even after downgrade to Cabal-1.23.1.0 and cabal-install-1.23.0.0, but worked after further downgrade -- to cabal-install 1.22.9.0 and Cabal 1.22.5.0.

**Comment by paul-rouse** [comment]
Fixed via #22 in version 0.1.2
